apps/passport-client/src/dispatch.ts

```diff
@@ -445,7 +469,7 @@ async function finishAccountCreation(
   if (hasPendingRequest()) {
     window.location.hash = "#/login-interstitial";
   } else {
-    window.location.hash = "#/";
+    window.location.hash = targetFolder ? `#/?folder=${targetFolder}` : "#/";
```

Comment:
```suggestion
    window.location.hash = targetFolder ? `#/?folder=${encodeURIComponent(targetFolder)}` : "#/";
```

also suggest omitting the `folder` query string param in the case `targetFolder` is null/undefined